src/dmd/audit_log.d

```diff
@@ -0,0 +1,81 @@
+/**
+ * Generate text files for assistance in debugging dmd.
+ *
+ */
+
+module dmd.audit_log;
+import core.stdc.stdlib;
+import core.stdc.stdio;
+import dmd.root.rootobject;
+import dmd.globals;
+import dmd.ast_node;
+import dmd.hdrgen;
+import dmd.expression;
+import dmd.dsymbol;
+import dmd.statement;
+import dmd.mtype;
+import dmd.init;
+import dmd.dtemplate;
+import dmd.common.outbuffer;
+
+private FILE*[string] file;
+
+void Audit_Log(ASTNode node, string functioncall = __FUNCTION__, string modulecall = __MODULE__)
+{
+    debug(AUDITLOG)
+    {
+        bool Exist()
+        {
+            for(int x = 0; x < global.params.calls.length; x++)
+            {
+                if(global.params.calls[x].toString() == functioncall
+                   || global.params.calls[x].toString() ==  modulecall)
+                    return true;
+            }
+            return false;
+        }
+        if(!Exist())
+            return;
+        auto x = "debug/" ~ modulecall ~ "/" ~ functioncall ~ ".txt\0";
+        auto y = fopen(x.ptr, "w+\0".ptr);
+        file[modulecall ~ " " ~ functioncall] = y;
```

Comment:
Since the log file is unconditionally `fopen`ed on every call to this function, the hash table doesn't actually save us any work. If we want to cache file handles, we should check for the file in the hash table *before* calling `fopen`.

Also, please use meaningful variable names instead of `x` and `y`.